src/components.jl

```diff
@@ -52,75 +52,81 @@ function coords(shapes::AVec{<:Shape})
 
 "interleave 2 vectors into each other (like a zipper's teeth)"
 function weave(x, y; ordering=Vector[x, y])
-  ret = eltype(x)[]
-  done = false
-  while !done
-    for o in ordering
-      try
-          push!(ret, popfirst!(o))
-      catch
-      end
+    ret = eltype(x)[]
+    done = false
+    while !done
+        for o in ordering
+            try
+                push!(ret, popfirst!(o))
+            catch
+            end
+        end
+        done = isempty(x) && isempty(y)
     end
-    done = isempty(x) && isempty(y)
-  end
-  ret
+    ret
 end
 
 "create a star by weaving together points from an outer and inner circle.  `n` is the number of arms"
 function makestar(n; offset=-0.5, radius=1.0)
     z1 = offset * π
     z2 = z1 + π / (n)
-    outercircle = partialcircle(z1, z1 + 2π, n+1, radius)
-    innercircle = partialcircle(z2, z2 + 2π, n+1, 0.4radius)
+    outercircle = partialcircle(z1, z1 + 2π, n + 1, radius)
+    innercircle = partialcircle(z2, z2 + 2π, n + 1, 0.4radius)
     Shape(weave(outercircle, innercircle))
 end
 
 "create a shape by picking points around the unit circle.  `n` is the number of point/sides, `offset` is the starting angle"
-makeshape(n; offset=-0.5, radius=1.0) = Shape(
-  partialcircle(offset * π, offset * π + 2π, n+1, radius)
-)
+makeshape(n; offset=-0.5, radius=1.0) =
+    Shape(partialcircle(offset * π, offset * π + 2π, n + 1, radius))
 
 function makecross(; offset=-0.5, radius=1.0)
     z2 = offset * π
-    z1 = z2 - π/8
+    z1 = z2 - π / 8
     outercircle = partialcircle(z1, z1 + 2π, 9, radius)
     innercircle = partialcircle(z2, z2 + 2π, 5, 0.5radius)
-    Shape(weave(outercircle, innercircle,
-                ordering=Vector[outercircle, innercircle, outercircle]))
+    Shape(
+        weave(
+            outercircle, innercircle; ordering=Vector[outercircle, innercircle, outercircle]
+        ),
+    )
 end
 
-from_polar(angle, dist) = P2(dist*cos(angle), dist*sin(angle))
+from_polar(angle, dist) = P2(dist * cos(angle), dist * sin(angle))
 
 makearrowhead(angle; h=2.0, w=0.4, tip=from_polar(angle, h)) = Shape(
-  P2[(0, 0), from_polar(angle - 0.5π, w) - tip, from_polar(angle + 0.5π, w) - tip, (0, 0)]
+    P2[
+        (0, 0),
+        from_polar(angle - 0.5π, w) - tip,
+        from_polar(angle + 0.5π, w) - tip,
+        (0, 0),
+    ],
 )
 
 const _shapes = KW(
-    :circle    => makeshape(20),
-    :rect       => makeshape(4, offset=-0.25),
-    :diamond    => makeshape(4),
-    :utriangle  => makeshape(3, offset=0.5),
-    :dtriangle  => makeshape(3, offset=-0.5),
-    :rtriangle  => makeshape(3, offset=0.0),
-    :ltriangle  => makeshape(3, offset=1.0),
-    :pentagon   => makeshape(5),
-    :hexagon    => makeshape(6),
-    :heptagon   => makeshape(7),
-    :octagon    => makeshape(8),
-    :cross      => makecross(offset=-0.25),
-    :xcross     => makecross(),
-    :vline      => Shape([(0, 1), (0, -1)]),
-    :hline      => Shape([(1, 0), (-1, 0)]),
-  )
```

Comment:
should keep the alignment here

Reply:
ok, should be the setting `align_pair_arrow`